overlord/hookstate/ctlcmd/set.go

```diff
@@ -113,6 +115,31 @@ func (s *setCommand) setConfigSetting(context *hookstate.Context) error {
 	return nil
 }
 
+func setInterfaceAttribute(context *hookstate.Context, staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, value interface{}) error {
+	data, err := json.Marshal(value)
+	if err != nil {
+		return fmt.Errorf("cannot marshal snap %q option %q: %s", context.SnapName(), key, err)
+	}
+	raw := json.RawMessage(data)
+
+	subkeys, err := config.ParseKey(key)
+	if err != nil {
+		return err
+	}
+
+	var existing interface{}
+	// We're called from setInterfaceSetting, subkeys is derived from key
+	// part of key=value argument and is guaranteed to be non-empty at this
+	// point.
+	err = config.GetFromChange(context.SnapName(), subkeys[:1], 0, staticAttrs, &existing)
+	if err == nil {
```

Comment:
we need to return the error if not a NoOption, no?

Reply:
Good point, done.